test/extended/networking/livemigration.go

```diff
@@ -0,0 +1,791 @@
+	for _, pod := range pods {
+		Eventually(func(g Gomega) {
+			var err error
+			pod, err = fr.ClientSet.CoreV1().Pods(fr.Namespace.Name).Get(context.TODO(), pod.Name, metav1.GetOptions{})
+			g.Expect(err).ToNot(HaveOccurred())
+			conditionFn(g, pod)
+		}).
+			WithTimeout(time.Minute).
+			WithPolling(time.Second).
+			Should(Succeed())
+	}
+}
+
+func generateAgnhostPod(name, namespace, nodeName string, args ...string) *corev1.Pod {
+	agnHostPod := e2epod.NewAgnhostPod(namespace, name, nil, nil, nil, args...)
+	agnHostPod.Spec.NodeName = nodeName
+	return agnHostPod
+}
+
+func createHTTPServerPods(fr *framework.Framework, annotations map[string]string, selectedNodes []corev1.Node) []*corev1.Pod {
+	var pods []*corev1.Pod
+	for _, selectedNode := range selectedNodes {
+		pod := generateAgnhostPod(
+			"testpod-"+selectedNode.Name,
+			fr.Namespace.Name,
+			selectedNode.Name,
+			"netexec", "--http-port", "8000")
+		pod.Annotations = annotations
+		pods = append(pods, e2epod.NewPodClient(fr).CreateSync(context.TODO(), pod))
+	}
+	return pods
+}
+
+func updatePods(fr *framework.Framework, pods []*corev1.Pod) []*corev1.Pod {
+	for i, pod := range pods {
+		var err error
+		pod, err = fr.ClientSet.CoreV1().Pods(fr.Namespace.Name).Get(context.TODO(), pod.Name, metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		pods[i] = pod
+	}
+	return pods
+}
+
+func podNetworkStatus(pod *v1.Pod, predicates ...func(nadapi.NetworkStatus) bool) ([]nadapi.NetworkStatus, error) {
+	podNetStatus, found := pod.Annotations[nadapi.NetworkStatusAnnot]
+	if !found {
+		return nil, fmt.Errorf("the pod must feature the `networks-status` annotation")
+	}
+
+	var netStatus []nadapi.NetworkStatus
+	if err := json.Unmarshal([]byte(podNetStatus), &netStatus); err != nil {
+		return nil, err
+	}
+
+	var netStatusMeetingPredicates []nadapi.NetworkStatus
+	for i := range netStatus {
+		for _, predicate := range predicates {
+			if predicate(netStatus[i]) {
+				netStatusMeetingPredicates = append(netStatusMeetingPredicates, netStatus[i])
+				continue
+			}
+		}
+	}
+	return netStatusMeetingPredicates, nil
+}
+func checkPodRunningReady() func(Gomega, *corev1.Pod) {
+	return func(g Gomega, pod *corev1.Pod) {
+		ok, err := testutils.PodRunningReady(pod)
+		g.Expect(err).ToNot(HaveOccurred())
+		g.Expect(ok).To(BeTrue())
+	}
+}
+
+func checkPodHasIPsAtNetwork(netName string, expectedNumberOfAddresses int) func(Gomega, *corev1.Pod) {
+	return func(g Gomega, pod *corev1.Pod) {
+		GinkgoHelper()
+		By(fmt.Sprintf("Checking pod annotations: %+v", pod.Annotations))
+		netStatus, err := podNetworkStatus(pod, func(status nadapi.NetworkStatus) bool {
+			return status.Name == netName
+		})
+		g.Expect(err).ToNot(HaveOccurred())
+		g.Expect(netStatus).To(HaveLen(1))
+		g.Expect(netStatus[0].IPs).To(HaveLen(expectedNumberOfAddresses))
+	}
+}
+
+func prepareHTTPServerPods(fr *framework.Framework, netConfig networkAttachmentConfigParams, selectedNodes []corev1.Node) []*corev1.Pod {
```

Comment:
For simplicity, if we use deployment  then we have to go a find the pods to look for IPs since pod name is generated, is simpler to just create a bunch of pods since we don't care for they lifecycle.